test/tests/ref.test.ts

```diff
@@ -0,0 +1,189 @@
+    const { refArrayNumber } = await RefTestModel.findById(refArrayId);
+    expect(refArrayNumber).to.deep.equal([_id1, _id2]);
+  });
+
+
+  it('check reference with buffer _id', async () => {
+    const id1 = Buffer.from([1, 2, 3, 4]);
+    const id2 = Buffer.from([5, 6, 7, 8]);
+
+    const refTypeTest = new RefTestModel();
+    refTypeTest.refFieldBuffer = id1;
+    refTypeTest.refArrayBuffer = [id1, id2];
+    refTypeTest.refFieldBuffer = new RefTestBufferModel();
+    refTypeTest.refArrayBuffer = [new RefTestBufferModel(), new RefTestBufferModel()];
+
+    const { _id: _id1 } = await new RefTestBufferModel({ _id: id1 }).save();
+    expect(_id1.equals(id1)).to.equal(true);
+    const { _id: _id2 } = await new RefTestBufferModel({ _id: id2 }).save();
+    expect(_id2.equals(id2)).to.equal(true);
+
+    const { _id: refBufferId } = await new RefTestModel({ refFieldBuffer: _id1 }).save();
+    const { refFieldBuffer } = await RefTestModel.findById(refBufferId);
+    expect(_id1.equals(refFieldBuffer)).to.equal(true);
+    const { _id: refArrayId } = await new RefTestModel({ refArrayBuffer: [_id1, _id2] }).save();
+    const { refArrayBuffer } = await RefTestModel.findById(refArrayId);
+    expect(refArrayBuffer).to.deep.equal([_id1, _id2]);
+  });
+
+  it('check typeguards', async () => {
+    const idFields = new RefTestModel({
+      refField: mongoose.Types.ObjectId(),
+      refArray: [mongoose.Types.ObjectId()],
+      refFieldString: 'test1',
+      refArrayString: ['test1', 'test2'],
+      refFieldNumber: 1234,
+      refArrayNumber: [1234, 5678],
+      refFieldBuffer: Buffer.from([1, 2, 3, 4]),
+      refArrayBuffer: [Buffer.from([1, 2, 3, 4]), Buffer.from([5, 6, 7, 8])]
+    });
+
+    expect(isDocument(idFields.refField)).to.equal(false);
+    expect(isDocument(idFields.refFieldString)).to.equal(false);
+    expect(isDocument(idFields.refFieldNumber)).to.equal(false);
+    expect(isDocument(idFields.refFieldBuffer)).to.equal(false);
+
+    expect(isDocumentArray(idFields.refArray)).to.equal(false);
+    expect(isDocumentArray(idFields.refArrayString)).to.equal(false);
+    expect(isDocumentArray(idFields.refArrayNumber)).to.equal(false);
+    expect(isDocumentArray(idFields.refArrayBuffer)).to.equal(false);
+
+    const populatedFields = new RefTestModel({
+      refField: new RefTestModel(),
+      refArray: [new RefTestModel(), new RefTestModel()],
+      refFieldString: new RefTestStringModel({ _id: 'test' }),
+      refArrayString: [new RefTestStringModel({ _id: 'test' }), new RefTestStringModel({ _id: 'test' })],
+      refFieldNumber: new RefTestNumberModel({ _id: 1234 }),
+      refArrayNumber: [new RefTestNumberModel({ _id: 1234 }), new RefTestNumberModel({ _id: 1234 })],
+      refFieldBuffer: new RefTestBufferModel({ _id: Buffer.from([1, 2, 3, 4]) }),
+      refArrayBuffer: [new RefTestBufferModel({ _id: Buffer.from([1, 2, 3, 4]) }),
```

Comment:
(i suggest this change, because vscode on save somehow formats it wrong ...)
```suggestion
      refArrayBuffer: [
        new RefTestBufferModel({ _id: Buffer.from([1, 2, 3, 4]) }),
```